File: KalimaJavaExample/src/org/kalima/java/example/AlarmsCallback.java
package org.kalima.java.example;

import org.kalima.cache.lib.KMsg;
import org.kalima.kalimamq.message.KMessage;
import org.kalima.kalimamq.nodelib.MemCacheCallback;

public class AlarmsCallback implements MemCacheCallback {
	
	private String address;
	
	public AlarmsCallback(String address) {
		this.address = address;
	}

	@Override
	public String getAddress() {
		return address;
	}

	@Override
	public void putData(KMessage kMessage) {
		KMsg kMsg = KMsg.setMessage(kMessage);
		System.out.println("new alarm key=" + kMsg.getKey() + " body=" + new String(kMsg.getBody()));
	}

	@Override
	public void removeData(KMessage kMessage) {

	}
}

File: KalimaCSharpExample/KalimaCSharpExample/AlarmsCallback.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using org.kalima.kalimamq.message;
using org.kalima.kalimamq.nodelib;
using org.kalima.cache.lib;


namespace KalimaCSharpExample
{
    class AlarmsCallback : MemCacheCallback
    {

        private String address;

        public AlarmsCallback(String address)
        {
            this.address = address;
        }
        public string getAddress()
        {
            return address;
        }

        public void putData(string key, KMessage kMessage)
        {
            KMsg kMsg = KMsg.setMessage(kMessage);
            Console.WriteLine("new alarm key=" + key + " body=" + System.Text.Encoding.Default.GetString(kMsg.getBody()));
        }

        public void removeData(string str)
        {
           
        }
    }
}
